core/src/main/scala/org/bykn/bosatsu/NameKind.scala

```diff
@@ -2,38 +2,38 @@ package org.bykn.bosatsu
 
 import Identifier.Bindable
 
-sealed abstract class NameKind
+sealed abstract class NameKind[T]
```

Comment:
I think if you make `+T`, so T is covariant, then only `Let[T]` has a type parameter, the others can extend `NameKind[Nothing]` which may make some of the changes go away.

Reply:
oops, missed this. I can go back and fix this